### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by mtene** [albertziegenhagel/childdebugger-vscode#16] [General]
I just want to report that I was able to debug a complicated commercial console application running with MPI in Windows via this extension. It just needs to be better integrated into the VSCode UI - e.g. switching between processes via the debugging hover (where the stop, step, etc buttons are) instead of through the stack trace drop-downs. I wonder if this can be extended to Linux debugging (gdb, lldb) or is it specific to the Microsoft debugger?

**Comment by albertziegenhagel** [comment]
I am very happy that it works for you! Debugging MPI applications is actually the main reason I have developed this extension to begin with.

>  It just needs to be better integrated into the VSCode UI - e.g. switching between processes via the debugging hover (where the stop, step, etc buttons are) instead of through the stack trace drop-downs.

All the UI for multiple debug sessions and their parent/child relationship is unfortunately not controlled by this extension, but is built into VS Code itself. This means (to the best of my knowledge) UI changes would have to be made to the main VS Code repository.

> I wonder if this can be extended to Linux debugging (gdb, lldb) or is it specific to the Microsoft debugger?

The implementation of this extension is very specific to the vsdbg-engine, so unfortunately it is `cppvsdbg` only and extending it to gdb or lldb would require a completely new and different implementation. To the best of my knowledge gdb and lldb do not provide the possibility to develop plugins for the debug engine (but I could just not know about it), so I wouldn't even know how to start working on this. On the other hand, those debuggers are open-source so there might be much better approaches to tackling this problem. I think the debug adapter protocol does actually already support this (see https://microsoft.github.io/debug-adapter-protocol/specification#Reverse_Requests_StartDebugging).
